src/safeposix/syscalls/net_calls.rs

```diff
@@ -1242,6 +1242,108 @@ impl Cage {
+    ///
+    /// ### Returns
+    ///
+    /// On success, the call returns the number of bytes sent. On error, a
+    /// negative error number is returned, with the errorno set to represent
+    /// the corresponding error
+    ///
+    /// ### Errors
+    ///
+    /// These are some standard errors generated by the socket layer.
+    ///    Additional errors may be generated and returned from the
+    ///    underlying protocol modules; see their respective manual pages.
+    ///
+    /// * EACCES - (For UNIX domain sockets, which are identified by pathname)
+    ///   Write permission is denied on the destination socket file, or search
+    ///   permission is denied for one of the directories the path prefix.  (See
+    ///   path_resolution(7).)  (For UDP sockets) An attempt was made to send to
+    ///   a network/broadcast address as though it was a unicast address.
+    ///
+    /// ** EAGAIN or EWOULDBLOCK - The socket is marked nonblocking and the
+    ///   requested operation would block.  POSIX.1-2001 allows either error to
+    ///   be returned for this case, and does not require these constants to
+    ///   have the same value, so a portable application should check for both
+    ///   possibilities.
+    ///
+    /// ** EAGAIN - (Internet domain datagram sockets) The socket referred to by
+    ///   sockfd had not previously been bound to an address and, upon
+    ///   attempting to bind it to an ephemeral port, it was determined that all
+    ///   port numbers in the ephemeral port range are currently in use.  See
+    ///   the discussion of /proc/sys/net/ipv4/ip_local_port_range in ip(7).
+    ///
+    /// * EALREADY - Another Fast Open is in progress.
+    ///
+    /// ** EBADF - sockfd is not a valid open file descriptor.
+    ///
+    /// * ECONNRESET - Connection reset by peer.
+    ///
+    /// * EDESTADDRREQ - The socket is not connection-mode, and no peer address
+    ///   is set.
+    ///
+    /// * EFAULT - An invalid user space address was specified for an argument.
+    ///
+    /// * EINTR - A signal occurred before any data was transmitted; see
+    ///   signal(7).
+    ///
+    /// ** EINVAL - Invalid argument passed.
+    ///
+    /// * EISCONN - The connection-mode socket was connected already but a
+    ///   recipient was specified.  (Now either this error is returned, or the
+    ///   recipient specification is ignored.)
+    ///
+    /// * EMSGSIZE - The socket type requires that message be sent atomically,
+    ///   and the size of the message to be sent made this impossible.
+    ///
+    /// * ENOBUFS - The output queue for a network interface was full.  This
+    ///   generally indicates that the interface has stopped sending, but may be
+    ///   caused by transient congestion. (Normally, this does not occur in
+    ///   Linux.  Packets are just silently dropped when a device queue
+    ///   overflows.)
+    ///
+    /// * ENOMEM - No memory available.
```

Comment:
I don't think we return many of these in our code...

Reply:
Many of the errors still could be returned by the libc called so Nick and I agreed to leave them there but notate the ones that we could return.